extension/js/common/core/attachment.ts

```diff
@@ -64,6 +66,70 @@ export class Attachment {
     this.contentDescription = contentDescription || undefined;
   }
 
+  public static prepareFileAttachmentDownload = (attachment: Attachment, parentTabId: string) => {
+    const blacklistedFiles = [
+      '.ade',
+      '.adp',
+      '.apk',
+      '.appx',
+      '.appxbundle',
+      '.bat',
+      '.cab',
+      '.chm',
+      '.cmd',
+      '.com',
+      '.cpl',
+      '.diagcab',
+      '.diagcfg',
+      '.diagpack',
+      '.dll',
+      '.dmg',
+      '.ex',
+      '.ex_',
+      '.exe',
+      '.hta',
+      '.img',
+      '.ins',
+      '.iso',
+      '.isp',
+      '.jar',
+      '.jnlp',
+      '.js',
+      '.jse',
+      '.lib',
+      '.lnk',
+      '.mde',
+      '.msc',
+      '.msi',
+      '.msix',
+      '.msixbundle',
+      '.msp',
+      '.mst',
+      '.nsh',
+      '.pif',
+      '.ps1',
+      '.scr',
+      '.sct',
+      '.shb',
+      '.sys',
+      '.vb',
+      '.vbe',
+      '.vbs',
+      '.vhd',
+      '.vxd',
+      '.wsc',
+      '.wsf',
+      '.wsh',
+      '.xll',
+    ];
+    const badFileExtensionWarning = 'This executable file was not checked for viruses, and may be dangerous to download or run. Proceed anyway?'; // xss-safe-value
+    if (blacklistedFiles.some(badFileExtension => attachment.name.endsWith(badFileExtension))) {
+      BrowserMsg.send.showWarningForAttachmentDownload(parentTabId, { message: badFileExtensionWarning, attachment });
```

Comment:
Like this:
`if (! await BrowserMsg.send.showConfirmation(..........) {
  return;
}

Reply:
thank you - I just need to review the figure out how I can read the return value from the browserMsg with the help of your comment above.